js/infobar.js

```diff
@@ -0,0 +1,478 @@
+            }
+
+            self.hide();
+            try {
+                sessionStorage.setItem(self.prefName, 'true');
+            } catch (ex) {}
+        });
+
+        if (self.onshow.callback) {
+            self.onshow.callback();
+        }
+
+        if (config.opened) {
+            config.bar.hide();
+            config.bar.attr('aria-hidden', true);
+        } else {
+            config.bar.show();
+            config.bar.attr('aria-hidden', false);
+        }
+
+        return config.bar;
+    };
+
+    /**
+     * Hides the currently visible infoBar
+     */
+    InfoBar.prototype.hide = function () {
+        var self = this;
+        var target = (config.opened) ? self.element : config.bar;
+
+        target.animate({'height': 0}, 200, function () {
+            self.element.hide();
+            config.bar.attr('aria-hidden', true);
+        });
+    };
+
+    /**
+     * Populates the bar template with the relevant strings.
+     * @param {object} content - An object containing the template strings.
+     */
+    InfoBar.prototype.populateTempl = function(content) {
+        config.$message.text(content.msg);
+        config.$acceptButton.text(content.accept);
+        config.$cancelButton.text(content.cancel);
+    };
+
+    /**
+     * Returns true if the current UA is one of the user agents based
+     * on Firefox.
+     */
+    InfoBar.prototype.isLikeFirefox = function(userAgent) {
+        var ua = userAgent || navigator.userAgent;
+        return (/Iceweasel/i).test(ua) || (/IceCat/i).test(ua) ||
+            (/SeaMonkey/i).test(ua) || (/Camino/i).test(ua) ||
+            (/like Firefox/i).test(ua);
+    };
+
+    /**
+     * Rturns true if this is a version of Firefox mobile excluding Firefox for iOS
+     */
+    InfoBar.prototype.isFirefoxMobile = function(userAgent) {
+        var ua = userAgent || navigator.userAgent;
+        return /Mobile|Tablet|Fennec/.test(ua);
+    };
+
+    /**
+     * Returns true if this is *the* Fox
+     */
+    InfoBar.prototype.isFirefox = function(userAgent) {
+        var ua = userAgent || navigator.userAgent;
+        return (/\sFirefox/).test(ua) && !InfoBar.prototype.isLikeFirefox(ua);
+    }
+
+    /**
+     * Returns the users current Firefox version
+     */
+    InfoBar.prototype.getUserVersion = function(userAgent) {
+        var ua = userAgent || navigator.userAgent;
+        return parseInt(ua.match(/Firefox\/(\d+)/)[1], 10);
+    };
+
+    /**
+     * This implements the update bar, shown for Firefox desktop users that has a version
+     * installed two major versions older than the latest.
+     * @param latestVersion {string} - Latest major version of Firefox (optional)
+     * @param buildID {int} - The UA buildID (optional)
+     */
+    InfoBar.update = function(latestVersion, buildID) {
+
+        var content = {};
+        var latestVersion = parseInt(latestVersion || config.json.latestfx, 10);
+        var buildID = buildID || navigator.buildID;
```

Comment:
`latestVersion` and `buildID` shouldn't be `var`'d, as they are already named parameters.